lib/puppet/type/service.rb

```diff
@@ -60,6 +60,16 @@ module Puppet
         provider.manual_start
       end
 
+      # This only makes sense on systemd systems. Otherwise, it just defaults
+      # to disable.
+      newvalue(:mask, :event => :service_disabled) do
+        if provider.respond_to?(:mask)
+          provider.mask
+        else
+          provider.disable
+        end
+      end
```

Comment:
My worry here is that you now have to know more about your system that you really want to.

In the case that I put together, mask defaults to disabling the service, which is a good default.

Otherwise, you'd have to put logic in your code to work around the use of 'mask' needlessly, wouldn't you?

I'd like to keep this as is to reduce code usage complexity.